CONTRIBUTING.md

```diff
@@ -129,7 +129,7 @@ You will need basic `git` proficiency to contribute to
 manual. Type `git --help` in a shell and enjoy! If you prefer books, [Pro
 Git](https://git-scm.com/book/en/v2) is a very good reference.
 
-You'll need **[Python 3.8](https://github.com/huggingface/transformers/blob/main/setup.py#L426)** or above to contribute to 🤗 Transformers. Follow the steps below to start contributing:
+You'll need **[Python 3.8](https://github.com/huggingface/transformers/blob/main/setup.py#L426)** or above (but not 3.11 yet) to contribute to 🤗 Transformers. Follow the steps below to start contributing:
```

Comment:
I don't think this is correct

Pytorch docs state that for [linux, mac and windows ](https://pytorch.org/get-started/locally/), it is recommended to use python 3.8 - 3.11 i.e. inclusive of 3.11. It doesn't explicitly state incompatibility with 3.12 or above.

Transformers is also not exclusive to pytorch usage - some users run with tensorflow or jax

Reply:
Thanks for the quick answer, Amy. I had some issues running `pip install -e .[dev]` with python 3.11 (first related to cython, then with some dependencies) until I downgraded it to Python 3.10. 

One can reproduce it by running it in a docker:
```
FROM python:3.11

# Install system dependencies
RUN apt-get update && apt-get install -y git 

# Set the working directory
WORKDIR /usr/src/app

# Clone the transformers repository
RUN git clone https://github.com/huggingface/transformers.git

# Set the working directory to the cloned repository
WORKDIR /usr/src/app/transformers

# Install development dependencies
RUN pip install -e .[dev]
```
the error I got was:

```
5.80 Collecting av==9.2.0 (from transformers==4.43.0.dev0)
25.82   Downloading av-9.2.0.tar.gz (2.4 MB)
26.04      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 11.8 MB/s eta 0:00:00
26.22   Installing build dependencies: started
31.02   Installing build dependencies: finished with status 'done'
31.03   Getting requirements to build wheel: started
31.44   Getting requirements to build wheel: finished with status 'error'
31.45   error: subprocess-exited-with-error
31.45   
31.45   × Getting requirements to build wheel did not run successfully.
31.45   │ exit code: 1
31.45   ╰─> [11 lines of output]
31.45       Package libavformat was not found in the pkg-config search path.
31.45       Perhaps you should add the directory containing `libavformat.pc'
31.45       to the PKG_CONFIG_PATH environment variable
31.45       Package 'libavformat', required by 'virtual:world', not found
31.45       Package 'libavcodec', required by 'virtual:world', not found
31.45       Package 'libavdevice', required by 'virtual:world', not found
31.45       Package 'libavutil', required by 'virtual:world', not found
31.45       Package 'libavfilter', required by 'virtual:world', not found
31.45       Package 'libswscale', required by 'virtual:world', not found
31.45       Package 'libswresample', required by 'virtual:world', not found
31.45       pkg-config could not find libraries ['avformat', 'avcodec', 'avdevice', 'avutil', 'avfilter', 'swscale', 'swresample']
31.45       [end of output]
31.45   
31.45   note: This error originates from a subprocess, and is likely not a problem with pip.
31.45 error: subprocess-exited-with-error
```

Reply:
@manuelsh Thanks for the detailed explanation! Did you see the same issues when installing with `pip install -e .` ? i.e. not the `dev` option? cc @ydshieh

Reply:
No problem without the `[dev]` option. It runs smoothly. I tried also to change `av` package from ver 9.2.0 to 12.2.0 in the `python.py` file, but then it clashes with `jaxlib` package...

Reply:
OK, there's a wider issue here about compatibility with some of the libraries, but this is somewhat orthogonal to the python version. In this case, we can open an issue to track and try and find a set of libraries which work, but I don't think we should update the docs here as it can be misleading

Reply:
I agree, @amyeroberts. We can close the PR.